test/src/unit-filter.cc

```diff
@@ -0,0 +1,321 @@
+#include <iostream>
+
+using namespace tiledb::sm;
+
+TEST_CASE(
+    "Filter: Test bit width reduction filter deserialization",
+    "[filter][bit-width-reduction]") {
+  uint32_t max_window_size0 = 1024;
+  BitWidthReductionFilter filter0(max_window_size0);
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+
+  uint32_t max_window_size1 = 0;
+  REQUIRE(
+      filter1.value()
+          ->get_option(FilterOption::BIT_WIDTH_MAX_WINDOW, &max_window_size1)
+          .ok());
+  CHECK(max_window_size0 == max_window_size1);
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test bit shuffle filter deserialization",
+    "[filter][bit-shuffle]") {
+  BitshuffleFilter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test byte shuffle filter deserialization",
+    "[filter][byte-shuffle]") {
+  ByteshuffleFilter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test checksum md5 filter deserialization",
+    "[filter][checksum-md5]") {
+  ChecksumMD5Filter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test checksum sha256 filter deserialization",
+    "[filter][checksum-sha256]") {
+  ChecksumSHA256Filter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test encryption aes256gcm filter deserialization",
+    "[filter][encryption-aes256gcm]") {
+  EncryptionKey encryptionkey0;
+  encryptionkey0.set_key(
+      EncryptionType::AES_256_GCM, "abcdefghijklmnopqrstuvwxyz012345", 32);
+  EncryptionAES256GCMFilter filter0(encryptionkey0);
+
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
```

Comment:
check for equal keys as well?

Reply:
encryption key is not serialized and not stored on disk